### PR TITLE
Disable mint functionalities until Sigma is activated

### PR DESCRIPTION
**Pull request by ultimaweapon**

### CODE REVIEW

**Comment by thebevrishot** [comment]
please provide result image.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2714,6 +2714,16 @@ UniValue mint(const UniValue& params, bool fHelp)
             "\nAutomatically choose denominations to mint by amount\n" +
             HelpRequiringPassphrase());
 
+    // Ensure Sigma mints is already accepted by network so users will not lost their coins
+    // due to other nodes will treat it as garbage data.
+    {
+        LOCK(cs_main);
```
Comment (martun) [inline src/wallet/rpcwallet.cpp]: Is it necessary to lock cs_main here?

**Comment by ultimaweapon** [comment]
@a-bezrukov okay good idea. I agree.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (ultimaweapon) [reply src/wallet/rpcwallet.cpp]: Yes because we want to access `chainActive`.

File: src/zerocoin_v3.cpp [hunk 2]
```diff
@@ -51,6 +51,17 @@ static bool CheckZerocoinSpendSerialV3(
 	return true;
 }
 
+bool IsSigmaAllowed()
+{
+	LOCK(cs_main);
```
Comment (martun) [inline src/zerocoin_v3.cpp]: After we merge this one, this lock is not necessary any more right? https://github.com/zcoinofficial/zcoin/pull/373/files

Comment (ultimaweapon) [reply src/zerocoin_v3.cpp]: If we continue with the way in that PR, yes.

**Pull request merged**